dtao.c

```diff
@@ -345,13 +404,144 @@ static struct zwlr_layer_surface_v1_listener layer_surface_listener = {
 	.closed = layer_surface_closed,
 };
 
+struct input_state {
+	struct wl_surface *surface;
+	double x, y;
+	uint32_t button;
+};
+
+static void
+noop() {}
+
+static void
+wl_pointer_enter(void *data, struct wl_pointer *wl_pointer,
+			uint32_t serial, struct wl_surface *surface,
+			wl_fixed_t surface_x, wl_fixed_t surface_y)
+{
+	struct input_state *istate = data;
+	istate->surface = surface;
+	istate->x = wl_fixed_to_double(surface_x);
+	istate->y = wl_fixed_to_double(surface_y);
+}
+
+static void
+wl_pointer_motion(void *data, struct wl_pointer *wl_pointer, uint32_t time,
+		wl_fixed_t surface_x, wl_fixed_t surface_y)
+{
+	struct input_state *istate = data;
+	istate->x = wl_fixed_to_double(surface_x);
+	istate->y = wl_fixed_to_double(surface_y);
+}
+
+static void
+wl_pointer_button(void *data, struct wl_pointer *wl_pointer, uint32_t serial,
+		uint32_t time, uint32_t button, uint32_t state)
+{
+	struct input_state *istate = data;
+	istate->button = state == WL_POINTER_BUTTON_STATE_RELEASED ? 0 : button;
+}
+
+static void
+wl_pointer_leave(void *data, struct wl_pointer *wl_pointer,
+		uint32_t serial, struct wl_surface *surface)
+{
+	struct input_state *istate = data;
+	istate->surface = NULL;
+}
+
+/* taken from dzen's util.c */
+void
+spawn(const char *arg) {
+	static const char *shell = NULL;
+
+	if(!shell && !(shell = getenv("SHELL")))
+		shell = "/bin/sh";
+	if(!arg)
+		return;
+	/* The double-fork construct avoids zombie processes and keeps the code
+	* clean from stupid signal handlers. */
+	if(fork() == 0) {
+		if(fork() == 0) {
+			setsid();
+			execl(shell, shell, "-c", arg, (char *)NULL);
+			fprintf(stderr, "dtao: execl '%s -c %s'", shell, arg);
+			perror(" failed");
+		}
+		exit(0);
+	}
```

Comment:
@djpohly, doesn't like double forking[^1], instead use a simple fork and ignore SIGCHLD

[^1]: https://github.com/djpohly/dwl/pull/211#issuecomment-1081304389